src/components/NurseryWithdrawals/NurseryReassignment.tsx

```diff
@@ -1,13 +1,193 @@
+import useDeviceInfo from 'src/utils/useDeviceInfo';
+import strings from 'src/strings';
+import { APP_PATHS } from 'src/constants';
+import TfMain from 'src/components/common/TfMain';
+import TitleDescription from 'src/components/common/TitleDescription';
+import Card from 'src/components/common/Card';
+import FormBottomBar from 'src/components/common/FormBottomBar';
+import ReassignmentRenderer, { PlotsMap, ZonesMap } from './ReassignmentRenderer';
+
+const columns: TableColumnType[] = [
+  { key: 'species', name: strings.SPECIES, type: 'string' },
+  { key: 'plantingSite', name: strings.PLANTING_SITE, type: 'string' },
+  { key: 'zone', name: strings.ZONE, type: 'string' },
+  { key: 'originalPlot', name: strings.ORIGINAL_PLOT, type: 'string' },
+  { key: 'newPlot', name: strings.NEW_PLOT, type: 'string' },
+  { key: 'numPlants', name: strings.QUANTITY, type: 'number' },
+  { key: 'reassign', name: strings.REASSIGN, type: 'string' },
+  { key: 'notes', name: strings.NOTES, type: 'string' },
+];
 
 type NurseryReassignmentProps = {
   organization: ServerOrganization;
 };
 
 export default function NurseryReassignment(props: NurseryReassignmentProps): JSX.Element {
+  const { organization } = props;
+  const theme = useTheme();
+  const history = useHistory();
+  const { isMobile } = useDeviceInfo();
   const { deliveryId } = useParams<{ deliveryId: string }>();
+  const [snackbar] = useState(useSnackbar());
+  const [speciesMap, setSpeciesMap] = useState<{ [id: string]: string }>();
+  const [delivery, setDelivery] = useState<Delivery>();
+  const [plots, setPlots] = useState<PlotsMap>();
+  const [zones, setZones] = useState<ZonesMap>();
+  const [siteName, setSiteName] = useState<string>();
+
+  // populate map of species id to scientific name
+  useEffect(() => {
+    const populateSpecies = async () => {
+      const speciesResponse = await getAllSpecies(organization.id);
+      if (speciesResponse.requestSucceeded) {
+        setSpeciesMap(
+          speciesResponse.species.reduce((acc: any, current: any) => {
+            acc[current.id] = current.scientificName;
+            return acc;
+          }, {})
+        );
+      } else {
+        snackbar.toastError();
+      }
+    };
+
+    populateSpecies();
+  }, [organization.id, snackbar]);
+
+  // populate delivery
+  useEffect(() => {
+    if (!deliveryId) {
+      return;
+    }
+
+    const populateDelivery = async () => {
+      const response = await getDelivery(Number(deliveryId));
+      if (response.requestSucceeded) {
+        setDelivery(response.delivery);
+      } else {
+        snackbar.toastError(response.error);
+      }
+    };
+
+    populateDelivery();
+  }, [deliveryId, snackbar]);
+
+  // populate lookup maps of zones and plots
+  useEffect(() => {
+    if (!delivery) {
+      return;
+    }
+
+    const populatePlots = async () => {
+      const plotIds = delivery.plantings.map((planting) => planting.plotId).filter((id) => id);
```

Comment:
What is filter id=> id doing ?

Reply:
plantings can have undefined plotid, we filter only those ids that are valid

Reply:
Thanks!